float32 alpha
float32 betha
float32 gamma
